dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcCompletenessTableManager.java

```diff
@@ -256,7 +256,12 @@ private String getPartitionClause(AnalyticsTablePartition partition) {
 
   private List<AnalyticsTableColumn> getColumns() {
     String idColAlias = "concat(ds.uid,'-',ps.iso,'-',ous.organisationunituid,'-',ao.uid) as id ";
-    String timelyDateDiff = "cast(cdr.date as date) - ps.enddate";
+    int secondsInDay = (int) (24L * 60 * 60);
+
+    // The timeliness clock can only start ticking at the
+    String timelyDateDiff =
+        format(
+            "(EXTRACT(EPOCH FROM (cdr.date - ps.enddate)) - {}) / {}", secondsInDay, secondsInDay);
```

Comment:
```suggestion
            "(extract(epoch from (cdr.date - ps.enddate)) - {}) / {}", secondsInDay, secondsInDay);
```

Reply:
Even though I would prefer uppercase for SQL/Functions, our internal standard uses lowercase.